samples/net/sockets/http_client/src/main.c

```diff
@@ -353,5 +353,7 @@ void main(void)
 		close(sock6);
 	}
 
-	k_sleep(K_FOREVER);
+	while (true) {
```

Comment:
@nordic-krch just fixed k_sleep(K_FOREVER), so why do you need to make this change?

Reply:
No, he fixed it so if you pass in a negative duration it returns immediately, just like it did until yesterday.  #19748 needs to be fixed to avoid the premature return when the maximum timeout expires.

Reply:
Help me out here because I'm dumb. I am looking at #20448 and it fixes the behavior where k_sleep(K_FOREVER) returns immediately.

Reply:
Yes.  So what it does in that case is:

    ticks = (ms == K_FOREVER) ? INT32_MAX - _TICK_ALIGN) : k_ms_to_ticks_ceil32(ms);
    ticks = z_tick_sleep(ticks);

So what it's going to do is sleep for 2^31 - 1 - 1 = 2147483646 ticks.

At 32 kHz ticks, that means it will wake up in 65535.999938 seconds.

Which is about 18:12:16 hours:minutes:seconds.

Not forever.

When it returns, if it isn't in a loop, it falls off the end of the earth, which is probably bad.

I'm confused too, because in https://github.com/zephyrproject-rtos/zephyr/pull/20448#pullrequestreview-313909777 you correctly noted that for this to work as intended it has to be taken off the scheduling queue.

Reply:
let's address that then -- I have some time this morning

Reply:
OK, but it also probably applies to every other use of `K_FOREVER` in the baseline (`k_poll`, e.g).  I'm thinking it's a little hard to get that implemented, tested, and approved by tonight, though maybe it could be considered a bug fix and doesn't need to be resolved immediately.

Reply:
I'm inclined to treat this as a bug.
Thanks for the additional clarification, I didn't realize that the current proposed solution expired in hours, I wouldn't have +1'd it